### ISSUE TITLE
pyright 1.1.212 raises internal error: maximum call stack exceeded 

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#2902]
**Describe the bug**
Running pyright-basic on [this file](https://github.com/mit-ll-responsible-ai/hydra-zen/blob/fe64ae349f0a8b1ea9ad24fb70d53877c93d5ecb/src/hydra_zen/structured_configs/_implementations.py) produces the following internal error.
```
An internal error occurred while type checking file "C:\Users\Ryan Soklaski\hydra-zen\src\hydra_zen\structured_configs\_implementations.py": RangeError: Maximum call stack size exceeded
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1271:54)
    at Array.forEach (<anonymous>)
    at setTypeArgumentsRecursive (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1271:45)
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1247:21)
    at Array.forEach (<anonymous>)
    at setTypeArgumentsRecursive (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1246:40)
    at callback (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1240:17)
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:407:13)
    at Array.forEach (<anonymous>)
    at d (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:406:23)
Error performing analysis: RangeError: Maximum call stack size exceeded
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1271:54)
    at Array.forEach (<anonymous>)
    at setTypeArgumentsRecursive (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1271:45)
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1247:21)
    at Array.forEach (<anonymous>)
    at setTypeArgumentsRecursive (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1246:40)
    at callback (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:1240:17)
    at forEach (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:407:13)
    at Array.forEach (<anonymous>)
    at d (C:\Users\Ryan Soklaski\Anaconda3\envs\hydra-zen\node_modules\pyright\dist\pyright-internal\src\analyzer\typeUtils.ts:406:23)
```

**To Reproduce**
I am not sure how to narrow this down, unfortunately. But I confirmed that there is no such issue for pyright 1.1.211.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a long-standing bug that was previously masked, but a recent code change unmasked it. It will be fixed in the next release. Apologies for the inconvenience.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.213, which I just published. It will also be included in the next release of pylance.
